src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java

```diff
@@ -0,0 +1,270 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.data.cosmosdb.performance;
+
+import com.google.common.collect.Lists;
+import com.microsoft.azure.documentdb.DocumentClient;
+import com.microsoft.azure.documentdb.DocumentClientException;
+import com.microsoft.azure.spring.data.cosmosdb.core.query.DocumentDbPageRequest;
+import com.microsoft.azure.spring.data.cosmosdb.performance.domain.PerfPerson;
+import com.microsoft.azure.spring.data.cosmosdb.performance.repository.PerfPersonRepository;
+import com.microsoft.azure.spring.data.cosmosdb.performance.service.SdkService;
+import com.microsoft.azure.spring.data.cosmosdb.performance.utils.Constants;
+import com.microsoft.azure.spring.data.cosmosdb.performance.utils.DatabaseUtils;
+import com.microsoft.azure.spring.data.cosmosdb.performance.utils.PerfDataProvider;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.microsoft.azure.spring.data.cosmosdb.performance.utils.FunctionUtils.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = PerfConfiguration.class)
+public class PerformanceCompareIT {
+    @Autowired
+    private DocumentClient documentClient;
+
+    @Autowired
+    private PerfPersonRepository repository;
+
+    private static final int TIMES = 100;
+    private static final int BATCH_SIZE = 10;
+    private static final float ACCEPTANCE_DIFF = 0.1f;
+    private static boolean hasInit = false;
+    private static SdkService sdkService;
+    private static PerformanceReport report = new PerformanceReport();
+
+    @Before
+    public void setup() throws DocumentClientException {
+        if (!hasInit) {
```

Comment:
seems you may need `@BeforeClass` ? Please correct me if wrong.

Reply:
`@BeforeClass` requires static  method, which cannot use the Autowired non-static variable.